proto/cacheclient.proto

```diff
@@ -402,16 +402,110 @@ message _ListRemoveRequest {
 
 message _ListRemoveResponse {}
 
+/*
+  # ListFetch and ListRetain documentation and examples
+
+  # start and end are zero-based indexes, both are optional
+
+  # start is inclusive
+  # If start is not set, then the default is Unbounded, which is treated as 0
+  # Unbounded == 0,
+  # this field has a explicit presence for future compatibility to support reverse order and controlling our stride
+
+  # end is exclusive
+  # If end is not set, then the default is Unbounded which becomes (list.len)
+
+  # start and end can be negative numbers, negative indicates offsets starting at the end of the list.
+  # For example, -1 is the last element of the list, -2 the penultimate, etc
+  #               [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+  # +start/+end    0, 1, 2, 3, 4, 5, 6, 7, 8, 9
```

Comment:
What does "+start/+end" mean? Does it mean `0/9`? That would output 0,1...,7,8 though based on the preceding rule.

Oh.. do you mean
```suggestion
  # For example, -1 is the last element of the list, -2 the penultimate, etc
  # list:                     [  a,  b,  c,  d,  e,  f,  g,  h,  i,  j]
  # positive index positions:    0,  1,  2,  3,  4,  5,  6,  7,  8,  9
  # negative index positions:  -10, -9, -8, -7, -6, -5, -4, -3, -2, -1
```